src/components/Restoration/ApprovalType.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <div id="approval-type">
+    <div class="section-container" :class="{ 'invalid-section': invalidSection }">
+      <ApprovalTypeShared
+        :courtOrderNumber="getRestoration.courtOrder.fileNumber"
+        :approvedByRegistrar="approvedByRegistrar"
+        :noticeDate="getRestoration.noticeDate"
+        :applicationDate="getRestoration.applicationDate"
+        @radioButtonChange="setRestorationApprovalType($event)"
+        @courtNumberChange="setRestorationCourtOrder({ fileNumber: $event })"
+        @update:noticeDate="setRestorationNoticeDate($event)"
+        @update:applicationDate="setRestorationApplicationDate($event)"
+        @valid="setApprovalTypeValid($event)"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { ActionBindingIF, RestorationStateIF } from '@/interfaces'
+import { ApprovalType as ApprovalTypeShared } from '@bcrs-shared-components/approval-type'
+
+@Component({
+  components: {
+    ApprovalTypeShared
+  }
+})
+export default class ApprovalType extends Vue {
+  @Getter getApprovalTypeValid!: boolean
+  @Getter getRestoration!: RestorationStateIF
+  @Getter getShowErrors!: boolean
+
+  @Action setRestorationApprovalType!: ActionBindingIF
+  @Action setRestorationCourtOrder!: ActionBindingIF
+  @Action setRestorationNoticeDate!: ActionBindingIF
+  @Action setRestorationApplicationDate!: ActionBindingIF
+  @Action setApprovalTypeValid!: ActionBindingIF
+
+  /** This section's validity state (when prompted by app). */
+  get invalidSection (): boolean {
+    return (this.getShowErrors && !this.getApprovalTypeValid)
+  }
+
+  get approvedByRegistrar (): boolean {
+    return (!!this.getRestoration.noticeDate && !!this.getRestoration.applicationDate)
+  }
+}
+</script>
```

Comment:
@JazzarKarim This is a "shim" component between this app and your shared component. It's here that all store getters/actions are done, so that the parent component doesn't have to know about these things.

Reply:
Rather than passing a prop and event for each form field, what do you think of passing the entire "ApprovalTypeShared" object?  If we're separating concerns, the parent shouldn't have to know the fields contained within a shared component and should have to know when we add or remove a field inside the shared component.

Reply:
This thread continues in comments below.

Reply:
@jonathan-longe OK to resolve this conversation?